### DISCUSSION TITLE
Why not use decorators?

### DISCUSSION THREAD

**Discussion by axd1x8a** [fastapi/asyncer#135] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Asyncer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Asyncer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Asyncer but to [AnyIO](https://github.com/agronholm/anyio).
- [X] I already checked if it is not related to Asyncer but to [Trio](https://github.com/python-trio/trio).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import time
from typing import reveal_type

import anyio
from asyncer import asyncify


@asyncify
def do_sync_work(name: str):
    time.sleep(1)
    return f"Hello, {name}"


async def main():
    message = await do_sync_work(name="World")
    reveal_type(do_sync_work)  # Type of "do_sync_work" is "(name: str) -> Awaitable[str]"
    reveal_type(message)  # Type of "message" is "str"


anyio.run(main)
```

### Description

All `asyncer` functions work fine as decorators (and as I think the code looks better with them), so why is there no mention of this in the documentation?

### Operating System

Windows

### asyncer Version

0.0.5

### Python Version

Python 3.11.8

**Comment by tiangolo** [comment, marked as answer]
It's because decorating a function with `asyncify` would make it only awaitable, it wouldn't be callable from blocking code, while the logic and code underneath are purely sync. And documenting it could encourage it, while it's probably not ideal, so I prefer not to document it.
